packages/ui/ToggleItem/ToggleItem.js

```diff
@@ -15,26 +15,49 @@ import kind from '@enact/core/kind';
 import EnactPropTypes from '@enact/core/internal/prop-types';
 import PropTypes from 'prop-types';
 import compose from 'ramda/src/compose';
-import React from 'react';
+import React, {Fragment} from 'react';
 
 import ComponentOverride from '../ComponentOverride';
 import ForwardRef from '../ForwardRef';
+import Slottable from '../Slottable';
 import Toggleable from '../Toggleable';
 import Touchable from '../Touchable';
 
 import componentCss from './ToggleItem.module.less';
 
 // eslint-disable-next-line
-const iconCreator = (position) => ({disabled, icon, iconComponent, selected, iconPosition}) => {
-	if (iconPosition === position) {
+const iconCreator = (position) => ({css, disabled, icon, iconComponent, iconPosition, itemIcon, itemIconPosition, selected}) => {
+	if (position === 'before') {
 		return (
-			<ComponentOverride
-				component={iconComponent}
-				disabled={disabled}
-				selected={selected}
-			>
-				{icon}
-			</ComponentOverride>
+			<Fragment>
+				{itemIconPosition === 'before' && itemIcon && <ComponentOverride component={itemIcon} className={`${css.itemIcon} ${itemIcon.props.className}`} />}
```

Comment:
@ryanjduffy , Can I get the other approaches for mixing styling of framework with application's `className` ?

Reply:
It doesn't appear that you need to to because the `itemIcon` isn't defined anywhere. But, assuming that is an oversight, there isn't really another way other than reaching into its props. The challenge working with a React.Element (= the output of `<Component>` or `React.createElement(Component)`) is that it treated as a black box and only passed through other React methods (e.g. `cloneElement`). So while you can do something like the above now, it may not always work.

Other alternatives to solve the problem include:
* Document that `className` is passed to `itemIcon` and will override any specified value on the Element
* Only supporting `string|Object` for `itemIcon` which avoids the issue since the author has no means to specify a className
* Wrap the `itemIcon` with another node on which the class can be set
* Use a alternate attribute (e.g. `data-item-icon`) instead of a CSS class to use in the selector for styling

Each of these has its own trade offs which we can discuss.

Reply:
@Sangwook-Lee, how do you feel about consolidating this repeated code into a variable, then just refer to this variable in each of the four locations here and below?

Ex:
```
const itemIconComponent = (itemIcon ? <ComponentOverride component={itemIcon} className={`${css.itemIcon} ${itemIcon.props.className}`} /> : null);
...
{itemIconPosition === 'beforeText' && itemIconComponent}
{itemIconPosition === 'afterText' && itemIconComponent}
etc
```

Also, regarding the discussion from above, if `itemIcon` is a `kind()` component, shouldn't the classNames automatically merge in this case, rather than needing special handling? As in the following code example:
```
<ComponentOverride component={itemIcon} className={css.itemIcon} />
```
I've not tested in this regard, but this was my expectation based on what I recall in the past use cases. `iconComponent` of `SwitchItem.js` does this.

Reply:
@ryanjduffy , @Djspaceg , Thank you for the review. I've updated PR to use `kind()`. I think this issue was resolved by @Djspaceg 's suggestion.

Reply:
This would be a good candidate for a unit test to validate this assumption.